src/types.rs

```diff
@@ -11,14 +11,16 @@ use std::ops::Index;
 use pyo3::prelude::*;
 
 pub use custom::CustomType;
-pub use simple::{ClassicType, Container, SimpleType, TypeRow};
+pub use simple::{ClassicRow, ClassicType, Container, SimpleRow, SimpleType};
 
 use smol_str::SmolStr;
 
 use crate::hugr::{Direction, Port};
 use crate::utils::display_list;
 use crate::{resource::ResourceSet, type_row};
 
+use self::simple::{PrimType, TypeRow};
+
```

Comment:
Export `PrimType` and `TypeRow` too?

Reply:
So this is something I debate. Obviously Rust requires that these types have to be public, but I'd kinda like to encourage users to use the aliases. The case in which a user should be using TypeRow....almost certainly is in a function parameterized `<T: PrimType> ...TypeRow<T>....`

Which is not the common case, but it is a *reasonable* case ;-). So ok, exported (within `types` but no further, as nothing is exported out of `types`)

Reply:
However @aborgna-q one thing we might do is make PrimType sealed (i.e. can only be implemented inside the crate)....?